UnityProject/Assets/Scripts/Managers/GameManager.cs

```diff
@@ -291,6 +310,192 @@ public void ServerSetSpaceBody(MatrixMove mm)
 		PendingSpaceBodies.Enqueue(mm);
 	}
 
+	public static bool Is3D = false;
+	public GameObject objectToSpawn;
+
+	public void PlayerLoadedIn(NetworkConnectionToClient Player)
+	{
+		//TODO Action system Lazy for now
+
+		if (Is3D)
+		{
+			Activate3DMode.SendTo(Player);
+		}
+	}
+
+	public void PromptConvertTo3D()
+	{
+		if (Is3D) return;
+
+		DynamicChoiceUI.DisplayChoices("Doom? **Hold !!!TAB!!! to use mouse!!!***", " would you like to activate DOOM mode?, is a WIP so is buggy but is fun **Hold !!!TAB!!! to use mouse!!!***, Would you also like music to accompany it? ",
+			new List<DynamicUIChoiceEntryData>()
+			{
+				new DynamicUIChoiceEntryData()
+				{
+					ChoiceAction =  ConvertTo3DWithMusic,
+					Text = " Give me music and DOOM! ( Opens YouTube in browser )  "
+				},
+				new DynamicUIChoiceEntryData()
+				{
+					ChoiceAction =  ConvertTo3D,
+					Text = " Give me DOOM! "
+				},
+				new DynamicUIChoiceEntryData()
+				{
+					ChoiceAction = null,
+					Text = " No thank you  "
+				}
+			});
+	}
+
+
+	public void ConvertTo3DWithMusic()
+	{
+		Random rand = new Random();
+		int randomNumber = rand.Next(701);
+		char[] charArray = randomNumber.ToString().ToCharArray();
+		Array.Reverse(charArray);
+		string reversedString = new string(charArray);
+
+		if (reversedString == "007") //This never happened to the other fella
+		{
+			Application.OpenURL("https://youtu.be/AFaJWqVcv8k?t=11");
+		}
+		else
+		{
+			Application.OpenURL("https://youtu.be/0gEkNVq1ct0?t=8");
+		}
+
+		ConvertTo3D();
+	}
+
+
+
+	[NaughtyAttributes.Button()]
+	public void ConvertTo3D()
```

Comment:
I wish all of this wasn't in ``GameManager``. Why can't game manager make use of another class with this code contained in it? A lot cleaner than just slapping the joke mode into such an important class.

Reply:
3D manager?

Reply:
sure. Just move stuff over there and make ``GameManager`` use it.